### DISCUSSION TITLE
Do tablet buttons should work?

### DISCUSSION THREAD

**Discussion by Archirk** [wojtryb/Shortcut-Composer#76] [Help me]
First of all - great job. Amazing plugin. Perfect.

I faced strange behavior. May be it's not plugin but seems it could be.
Kubuntu 20.04, Krita 5.2.2, Wacom Tablet (4 buttons on tablet two buttons on stylus)

Lets say in system settings i maped button 1 of table to input "Ctrl+1"
I open Krita settings, choose pie-menu shortcut, changing input by pressing button1, "Ctrl+1" assign to pie menu. I apply.
In editor, pressing button 1 triggers only eye dropper(ctrl), but if I press Ctrl+1 on keyboard everything works.

Why I guess it's plugin issue - i in same way I assign native Krita action, e.g. "Add group layer" - it works.

Is it expected? Can i resolve it somehow.

**Comment by wojtryb** [comment]
You're doing a lot of tricky things here - I wouldn't really expect it to work well.

- pie-menus are quite non-standard actions, as they perform tasks both on key press and release. Tablet driver, which is emulating the keyboard, is probably not designed to handle events like that. If "ctrl+1" emits a key release event for ctrl before you release the button, the pie will close before it can fully open.
- in fact you can't even be sure if your tablet driver fully handles any key release events - it can just ignore holding it, and emit the release event right away
- key '1' is a canvas input for resettling a zoom. It's handled differently by krita and may create some clashes. You could unassign it in `settings->canvas inputs`

You can try to map your tablet button to something simpler like "a" and then map the pie to this key.

**Reply by Archirk** [reply]
Thanks for the swift reply. Sounds likely it's about press/release interptreted by driver. 

I will look into that if ubuntu driver has any settings about it. Because i vaguely remember familliar issues on windows when i tried to tweak something in Photoshop(probably also some settings to change brush sizes with cursor motion) and I has to choose special settings in wacom proprietary "Wacom Desktop Center" to emulate pressing button instead of press and release.

So yes definitely not plugin.
I will come back with an answer if find out solution during next few day or two.
